src/net/transport.rs

```diff
@@ -236,22 +256,24 @@ mod e2e_tests {
 
     async fn make_processors(
         ids: [HelperIdentity; 3],
-        conf: Arc<[peer::Config; 3]>,
+        sockets: [TcpListener; 3],
+        server_conf: [ServerConfig; 3],
+        network_conf: Arc<NetworkConfig>,
     ) -> [Processor<Arc<HttpTransport>>; 3] {
-        join_all(ids.map(|id| {
-            let conf = Arc::clone(&conf);
-            async move {
-                let transport = HttpTransport::new(id, conf);
-                transport.bind().await;
-                Processor::new(transport, ids).await
```

Comment:
Note that this is not ensuring that the local helper appears first in `ids`, which was supposed to be the case, prior to the changes in this PR.